R/stats-mlm-tidiers.R

```diff
@@ -0,0 +1,99 @@
+#' @templateVar class mlm
+#' @template title_desc_tidy
+#'
+#' @param x An `mlm` object created by [stats::lm()] with a matrix as the
+#'   response.
+#' @template param_confint
+#' @template param_quick
+#' @template param_unused_dots
+#'
+#' @evalRd return_tidy(regression = TRUE)
+#'
+#' @details In contrast to `lm` object (simple linear model), tidy output for
+#'   `mlm` (multiple linear model) objects contain an additional column
+#'   `response`.
+#'
+#'   If you have missing values in your model data, you may need to refit
+#'   the model with `na.action = na.exclude`.
+#'
+#' @examples
+#'
+#' mod <- lm(cbind(mpg, disp) ~ wt, mtcars)
+#' tidy(mod, conf.int = TRUE)
+#' @export
+#' @seealso [tidy()]
+#' @family lm tidiers
+#' @export
+
+tidy.mlm <- function(x,
+                     conf.int = FALSE,
+                     conf.level = .95,
+                     ...) {
+
+  # adding other details from summary object
+  s <- summary(x)
+  ret <- tidy.summary.mlm(s)
+
+  # adding confidence intervals
+  if (isTRUE(conf.int)) {
+
+    # S3 method for computing confidence intervals for `mlm` objects was
+    # introduced in R 3.5
+    CI <- tryCatch(
+      expr = stats::confint(x, level = conf.level),
+      error = function(x) {
+        NULL
+      }
+    )
+
+    # if R version is prior to 3.5, use the custom function
+    if (is.null(CI)) {
+      CI <- confint.mlm(x, level = conf.level)
+    }
+
+    colnames(CI) <- c("conf.low", "conf.high")
+    ret <- cbind(ret, unrowname(CI))
+  }
+
+  tibble::as_tibble(ret)
+}
+
+# mlm objects subclass lm objects so this gives a better error than
+# letting augment.lm() fail
+#' @include null-and-default-tidiers.R
+augment.mlm <- augment.default
+
+#' @export
+glance.mlm <- function(x, ...) {
+  stop(
+    "Glance does not support linear models with multiple responses.",
+    call. = FALSE
+  )
+}
+
+#' @export
+tidy.summary.mlm <- function(x, ...) {
+  co <- stats::coef(x)
+  nn <- c("estimate", "std.error", "statistic", "p.value")
+
+  # multiple response variables
+  ret <- purrr::map_df(co, fix_data_frame, nn[1:ncol(co[[1]])],
+    .id = "response"
+  )
+  ret$response <- stringr::str_replace(ret$response, "Response ", "")
+
+  as_tibble(ret)
+}
+
+# compute confidence intervals for mlm object.
+confint.mlm <- function(object, level = 0.95, ...) {
```

Comment:
Oh I see it made it back down here. I'm happy to keep this, or alternatively to just let this be an error. If you want to support R < 3.5, instead of check if the CI is `NULL`, this should just be directly in the catch of `tryCatch`.

Reply:
Hmm, unfortunately, I don't have much experience with `tryCatch` so I am not sure how to refactor this in a way that satisfied your requirements.

Reply:
No problem, I'll play around with it.